v2/pkg/engine/datasource/httpclient/nethttpclient.go

```diff
@@ -216,12 +220,12 @@ func makeHTTPRequest(client *http.Client, ctx context.Context, url, method, head
 		Request: TraceHTTPRequest{
 			Method:  request.Method,
 			URL:     request.URL.String(),
-			Headers: redactHeaders(request.Header),
+			Headers: redactHeaders(request.Header.Clone()),
```

Comment:
For safety purposes, I thought we should clone the headers, since we're also exposing them through the hooks

Reply:
They are already copied by `json.Marshal`